lib/membrane_rtc_engine/engine.ex

```diff
@@ -994,80 +959,73 @@ defmodule Membrane.RTC.Engine do
       end
     ]
 
-    endpoints_to_link =
-      ctx.children
-      |> Enum.flat_map(fn
-        {{:endpoint, endpoint_id}, _data} -> [endpoint_id]
-        _child -> []
-      end)
-      |> Enum.filter(&MapSet.member?(state.waiting_for_linking[&1], track_id))
+    {pending_subscriptions, rest} =
+      Enum.split_with(state.pending_subscriptions, fn s -> s.track_id == track.id end)
```

Comment:
I would change these names to maybe something like `pending_track_subscriptions` and `pending_rest_subscriptions`.